app/assets/javascripts/discourse/app/routes/discourse.js

```diff
@@ -92,7 +92,10 @@ const DiscourseRoute = Route.extend({
   },
 
   isPoppedState(transition) {
-    return !transition._discourse_intercepted && !!transition.intent.url;
+    return (
+      (!transition._discourse_intercepted && !!transition.intent.url) ||
+      (transition.sequence > 1 && transition.urlMethod)
```

Comment:
This is kind of a black box, I don't quite understand what we're checking here to determine popped state. Is there any chance we can add a descriptive comment?

Reply:
I am not familiar with all the inner working either, but it seems there is no other "proper" way of checking if user pressed the back button. I found this method here: https://github.com/emberjs/ember.js/issues/3087#issuecomment-22058755

Reply:
Yeah, tricky. I'm noticing that `transition.urlMethod` will return a value in many normal, non-back navigation transitions. So this change will return true for `isPoppedState` in many more cases, I suspect that will have undesirable effects. 

Maybe we can try something different: 

```suggestion
    return (
      !transition._discourse_intercepted &&
      (!!transition.intent.url || !!transition.queryParamsOnly)
    );
```

From a few quick tests locally, I am noticing that `transition.intent.url` is `undefined` when using the back button and landing on a topic list with query parameters. So adding a check for `queryParamsOnly` seems to help. And since we're still checking for `_discourse_intercepted` first in all cases, this should target only those topics lists more accurately. 

What do you think @nbianca?

Reply:
In which cases does it misdetect the normal transitions? I think it is related to how we transition between routes.

Anyway, your solution looks good to me and I added a few more fixes to follow up after my latest commit.